### DISCUSSION TITLE
[Feature Request] Modify sort order of list passed to the Multi Rename Tool

### DISCUSSION THREAD

**Discussion by DavidPesticcio** [doublecmd/doublecmd#1156] [Suggestions]
If this is currently possible, could someone please point me in the right direction... :smiley: 

It appears to me that the Multi Rename Tool (`CTRL-M`) is currently unable to affect the order of the list passed to it.

It would be useful to be able to "sort" the list in the Multi Rename Tool window to some arbitrary order - this would be useful when applying the `[C] Counter` placeholder to apply a range (or any other range function) to the list. Especially if the sort order cannot be accommodated by the sort function of the main window.

Perhaps initially adding the ability to reverse the order of the `[C] Counter` applied to the listed files would be a feature people could make use of. - Again, if this is currently possible, could someone please point me in the right direction... :smiley: 

I relalise this may be easier/better to script using a file that containes the list in the preffered arbitrary order... but ya know, if you don't ask and all that :grin:

Happy to help wherever I can... :smiley: :+1:

**Comment by Skif-off** [comment]
> Perhaps initially adding the ability to reverse the order of the [C] Counter applied to the listed files would be a feature people could make use of.

`Start Number = 100` and `Interval = -1`?

**Comment by DavidPesticcio** [comment]
Yeah, I tried that... let me paint a picture for you :smile: 

I have a list of 15 files in ascending date order `YYYY-MM`
I renamed 5 of them in the right sequence leading with `01-05` by hand - then remembered the Multi Rename Tool!
I then went to apply `06-15` to the remaining 10 with the Multi Rename Tool in the same way, but as I had manually renamed the initial 5 - your suggestion, and what I had already tried didn't help in this instance due to starting at 06 but needing to be applied in reverse to an ascending list order.

Although! Now that you've posed the suggestion, and I've thought about it, I realise that I should have reversed the list in the main window, ie. descending, before using the Multi Rename Tool, so that the order of the `[C] Counter` which was ascending would be just fine as I could then start the sequence at 06 to a descending list order and all with be well in the state of Denmark. (I've just done this, hence this additional piece of wisdom that solved my problem!)

My feature request was more aimed at an order that was less compliant - ie. a list that contains words that while they have a natural sort order, the required order may differ arbitrarily - like the names of chapters in a book, or sections of a manual for example, if these were stored in files of the same name as the chapter or section.

For convenience though, you may want to force the order by their chapter number rather than their "natural" order - which is the main reason for the feature request.

So the while the way you've suggested (Thanks! :smile:), plus some brain engagement from myself, actually solved my current problem, it would not be a fix in the above scenario... :crying_cat_face:

**Comment by Skif-off** [comment]
As I see, sorting (by clicking on the header(s)) is not supported, but drag&drop and Shift+arrows work fine (this is a little, but may be enough for a slight change the order of names :))

I'll try to check the help files and try to add it, if not described.

**Comment by DavidPesticcio** [comment]
Amazing! - exactly what I was looking for! :hugs::+1: 

Unfortunately, currently, there is no mention of `Drag & Drop` or of using `Shift+arrows` in the Multi Rename Tool documentation to facilitate changing the order of items in the list, nothing in the FAQ either... :slightly_frowning_face: 

If you could add something relevant to both that would be fantastic! :smile: :tada: 

https://doublecmd.github.io/doc/en/multirename.html
https://doublecmd.github.io/doc/en/faq.html

**Comment by Skif-off** [comment]
[PR#49](https://github.com/doublecmd/doublecmd.github.io/pull/49)
